src/widget/OktaSignIn.js

```diff
@@ -7,21 +7,22 @@ import buildRenderOptions from 'widget/buildRenderOptions';
 import createRouter from 'widget/createRouter';
 import V1Router from 'LoginRouter';
 import V2Router from 'v2/WidgetRouter';
+import Hooks from 'models/Hooks';
 
 const EVENTS_LIST = ['ready', 'afterError', 'afterRender'];
 
 var OktaSignIn = (function() {
 
   var router;
 
-  function getProperties(authClient, Router, widgetOptions = {}) {
+  function getProperties(authClient, hooks, Router, widgetOptions = {}) {
     // Returns a promise that will resolve on success or reject on error
     function render(renderOptions, successFn, errorFn) {
       if (router) {
         throw new Error('An instance of the widget has already been rendered. Call remove() first.');
       }
 
-      const res = createRouter(Router, widgetOptions, renderOptions, authClient, successFn, errorFn);
+      const res = createRouter(Router, widgetOptions, renderOptions, authClient, successFn, errorFn, hooks);
```

Comment:
hooks are basically widget options. It seems it can be part of `widgetOptions` instead of adding a new argument

Reply:
widgetOptions is the widget config passed to constructor. Although hooks can be defined here, they can also be added at runtime.